base/float.jl

```diff
@@ -688,22 +688,24 @@ function hash(x::Real, h::UInt)
     den_z = trailing_zeros(den)
     den >>= den_z
     pow += num_z - den_z
-
-    # handle values representable as Int64, UInt64, Float64
+    # If the real is an Int64, UInt64, or Float64, hash as those types.
+    # To be an Integer the denominator must be 1 and the power must be positive.
     if den == 1
+        # left = ceil(log2(num)*2^pow)
```

Comment:
```suggestion
        # left = ceil(log2(num*2^pow))
```
idk how much value this comment adds, but if we keep it, let's fix this